diff --git a/Makefile b/Makefile
index 886d99657..9705ee91b 100644
--- a/Makefile
+++ b/Makefile
@@ -507,7 +507,7 @@ LINUXINCLUDE    := \
 		-I$(objtree)/include \
 		$(USERINCLUDE)
 
-KBUILD_AFLAGS   := -D__ASSEMBLY__ -fno-PIE
+KBUILD_AFLAGS   := -g -D__ASSEMBLY__ -fno-PIE
 KBUILD_CFLAGS   := -Wall -Wundef -Werror=strict-prototypes -Wno-trigraphs \
 		   -fno-strict-aliasing -fno-common -fshort-wchar -fno-PIE \
 		   -Werror=implicit-function-declaration -Werror=implicit-int \
diff --git a/arch/arm64/configs/lightbox_gki.fragment b/arch/arm64/configs/lightbox_gki.fragment
new file mode 100644
index 000000000..a0ddc4553
--- /dev/null
+++ b/arch/arm64/configs/lightbox_gki.fragment
@@ -0,0 +1,9 @@
+CONFIG_DEVTMPFS=y
+CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_FUNCTION_TRACER=y
+CONFIG_NET_9P=y
+CONFIG_NET_9P_VIRTIO=y
+CONFIG_9P_FS=y
+CONFIG_PCI=y
+CONFIG_VIRTIO_PCI=y
+CONFIG_CPUFREQ_DUMMY=y
diff --git a/build.config.lightbox b/build.config.lightbox
new file mode 100644
index 000000000..52a374841
--- /dev/null
+++ b/build.config.lightbox
@@ -0,0 +1,20 @@
+. ${ROOT_DIR}/${KERNEL_DIR}/build.config.common
+. ${ROOT_DIR}/${KERNEL_DIR}/build.config.aarch64
+
+DEFCONFIG=lightbox_gki_defconfig
+PRE_DEFCONFIG_CMDS="cat ${ROOT_DIR}/${KERNEL_DIR}/arch/arm64/configs/gki_defconfig ${ROOT_DIR}/${KERNEL_DIR}/arch/arm64/configs/lightbox_gki.fragment > ${ROOT_DIR}/${KERNEL_DIR}/arch/arm64/configs/${DEFCONFIG};"
+POST_DEFCONFIG_CMDS="rm ${ROOT_DIR}/${KERNEL_DIR}/arch/arm64/configs/${DEFCONFIG}"
+POST_KERNEL_BUILD_CMDS="/bin/cp -r --remove-destination ${ROOT_DIR}/common/scripts/gdb ${ROOT_DIR}/out/android12-5.10/common/scripts/;ln -sf ${ROOT_DIR}/out/android12-5.10/common/scripts/gdb/vmlinux-gdb.py ${ROOT_DIR}/out/android12-5.10/common/vmlinux-gdb.py"
+
+MAKE_GOALS="${MAKE_GOALS}
+Image.lz4
+scripts_gdb
+"
+
+FILES="${FILES}
+arch/arm64/boot/Image.lz4
+"
+
+KMI_SYMBOL_LIST_STRICT_MODE=0
+
+LTO=none
